src/datasources/data-source.ts

```diff
@@ -0,0 +1,29 @@
+import { openmrsFetch, OpenmrsResource } from '@openmrs/esm-framework';
+import { DataSource } from '../api/types';
+
+export class BaseOpenMRSDataSource implements DataSource<any> {
+  private initialUrl: string;
+  private url: string;
+
+  constructor(url: string) {
+    this.initialUrl = url;
+    this.url = url;
+  }
+
+  fetchData(searchTerm: string, config?: Record<string, any>): Promise<any[]> {
+    this.url = this.initialUrl;
+    if (config?.tag) {
+      this.url = `${this.url}?tag=${config.tag}&v=custom:(uuid,display)`;
+    }
+    return openmrsFetch(searchTerm ? `${this.url}&q=${searchTerm}` : this.url).then(({ data }) => {
+      return data.results;
+    });
+  }
+
+  toUuidAndDisplay(data: any): OpenmrsResource {
```

Comment:
Is there any reason this isn't just:

```js
toUuidAndDisplay(data: OpenmrsResource): OpenmrsResource {
  return data;
}
```

I can maybe see it being useful for other kinds of resources.

Reply:
For us to do this, we need to make an assumption that it will be standard practice for anyone adding a datasource url to include `v=custom:(uuid,display)`

Reply:
This being the `BaseOpenmrsDataSource` it's a fair assumption. I agree with @ibacher

Reply:
@kajambiya my suggestion isn't introducing a _new_ assumption, i.e., this code already assumes that the `data` object has both a `uuid` and `display` fields. I don't really see how we can avoid making that assumption other than something like:

```js
if (typeof data.uuid === "undefined" || typeof data.display === "undefined") {
  throw new Error();
}
```